test/unit/CowWrapper.t.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Test} from "forge-std/Test.sol";
+import {CowWrapper, ICowSettlement, ICowAuthentication} from "../../src/CowWrapper.sol";
+import {EmptyWrapper} from "../EmptyWrapper.sol";
+
+import {MockWrapper, MockCowSettlement, MockCowAuthentication} from "./mocks/MockCowProtocol.sol";
+
+import {CowWrapperHelpers} from "../helpers/CowWrapperHelpers.sol";
+
+contract CowWrapperTest is Test {
+    MockCowAuthentication public authenticator;
+    MockCowSettlement public mockSettlement;
+    address public solver;
+
+    MockWrapper private wrapper1;
+    MockWrapper private wrapper2;
+    MockWrapper private wrapper3;
+
+    function setUp() public {
+        // Deploy mock contracts
+        authenticator = new MockCowAuthentication();
+        mockSettlement = new MockCowSettlement(address(authenticator));
+
+        solver = makeAddr("solver");
+        // Add solver to the authenticator
+        authenticator.setSolver(solver, true);
+
+        // Create test wrapper and three EmptyWrapper instances with the settlement contract
+        wrapper1 = new MockWrapper(ICowSettlement(address(mockSettlement)), 65536);
+        wrapper2 = new MockWrapper(ICowSettlement(address(mockSettlement)), 65536);
+        wrapper3 = new MockWrapper(ICowSettlement(address(mockSettlement)), 65536);
+
+        // Add all wrappers as solvers
+        authenticator.setSolver(address(wrapper1), true);
+        authenticator.setSolver(address(wrapper2), true);
+        authenticator.setSolver(address(wrapper3), true);
+    }
+
+    function _emptyInteractions() private pure returns (ICowSettlement.Interaction[][3] memory) {
+        return
+            [
+                new ICowSettlement.Interaction[](0),
+                new ICowSettlement.Interaction[](0),
+                new ICowSettlement.Interaction[](0)
+            ];
+    }
+
+    function _createSimpleSettleData(uint256 tokenCount) private returns (bytes memory) {
+        address[] memory tokens = new address[](tokenCount);
+        uint256[] memory clearingPrices = new uint256[](tokenCount);
+        for (uint256 i = 0; i < tokenCount; i++) {
+            tokens[i] = makeAddr(string(abi.encodePacked("Settle Token #", vm.toString(i + 1))));
+            clearingPrices[i] = 100 * (i + 1);
+        }
+        return abi.encodeWithSelector(
+            ICowSettlement.settle.selector, tokens, clearingPrices, new ICowSettlement.Trade[](0), _emptyInteractions()
+        );
+    }
+
+    function test_next_CallsWrapperAndThenNextSettlement() public {
+        bytes memory settleData = abi.encodePacked(_createSimpleSettleData(1), hex"123456");
```

Comment:
In a unit test this isn't really needed, right? You could just do settleData = encodePacked(0x13d79a0b, 0xanything)` if this helps removing a few lines of code from the tests.